assets/js/frontend/checkout.js

```diff
@@ -55,17 +55,21 @@ jQuery( function( $ ) {
 				$( 'input#createaccount' ).change( this.toggle_create_account ).change();
 			}
 		},
-		init_payment_methods: function() {
+		init_payment_methods: function( selectedPaymentMethod ) {
 			var $payment_methods = $( '.woocommerce-checkout' ).find( 'input[name="payment_method"]' );
 
 			// If there is one method, we can hide the radio input
 			if ( 1 === $payment_methods.length ) {
 				$payment_methods.eq(0).hide();
 			}
 
-			// If there are none selected, select the first.
-			if ( 0 === $payment_methods.filter( ':checked' ).length ) {
-				$payment_methods.eq(0).prop( 'checked', true );
+			if ( selectedPaymentMethod ) {
+				$( '#' + selectedPaymentMethod ).prop( 'checked', true );
+			} else {
+				// If there are none selected, select the first.
+				if ( 0 === $payment_methods.filter( ':checked' ).length ) {
```

Comment:
Sorry, 1 more thing :) This logic sees if there are checked elements, and if not checks the first.

Wouldn't it make sense to move this out of the conditonal `if ( selectedPaymentMethod ) {`? Then it can still run if the ID was not found for example.